walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,78 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    //If no attendees are able to attend
+    if(request.getAttendees().isEmpty() && events.isEmpty()){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    //If greater than lenght of day
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration()){
+        return Arrays.asList();
+    }
+
+    //return freetimes based on attendees
+    if(events.isEmpty() == false && request.getAttendees().isEmpty() == false){ //correct conditions?
```

Comment:
you can omit the == false and simply write

```java
if(!events.isEmpty() && !request.getAttendees().isEmpty()) {

}
```

Likewise, to check for true in if statements, you could write

```java
if (events.isEmpty()) {

}
```